app/schema/item.py

```diff
@@ -17,3 +15,48 @@ def construct_state(self):
 
     def get_state_class(self):
         pass
+
+    def validate(self, state):
+        if isinstance(state, self.get_state_class()):
+            is_valid = True
+            for child_state in state.children:
+                child_schema = self.questionnaire.get_item_by_id(child_state.id)
+                child_valid = child_schema.validate(child_state)
```

Comment:
possibly, but at present the state object doesn't hold a reference to it's associated schema counterpart.  This may change when we look at the augmentation strategy for rendering